hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/AuthenticationFuture.java

```diff
@@ -46,6 +47,6 @@ Connection get(int timeout) throws Throwable {
             return connection;
         }
         assert throwable != null;
-        throw throwable;
+        throw ExceptionUtil.rethrow(throwable);
```

Comment:
This doesn't fix the problem. THe original stacktrace still has no meaning for the receiver since its own stacktrace isn't 'added'.